### PR TITLE
doc: nrf: lib: modem_key_mgmt: Add code examples

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add code examples to show how to use the API.

### CODE REVIEW

File: doc/nrf/libraries/modem/modem_key_mgmt.rst [hunk 1]
```diff
@@ -57,6 +57,82 @@ The PEM format looks like this::
...
+   }
+
+The following code snippet shows how to deleta a CA chain certificate stored in the modem:
+
+.. code-block:: c
+
+   int err;
+
+   err = modem_key_mgmt_delete(<YOUR_SEC_TAG>, MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN);
+		if (err) {
```
Comment (MirkoCovizzi) [inline doc/nrf/libraries/modem/modem_key_mgmt.rst:133]: Indentation.

File: doc/nrf/libraries/modem/modem_key_mgmt.rst [hunk 2]
```diff
@@ -57,6 +57,82 @@ The PEM format looks like this::
...
+.. code-block:: c
+
+   int mismatch;
+   static const char cert[] = {
+           #include "YourCert.pem.inc"
+   };
+
+   mismatch = modem_key_mgmt_cmp(<YOUR_SEC_TAG>, MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, cert, sizeof(cert));
+   if (mismatch) {
+           printk("Certificate mismatch\n");
```
Comment (lemrey) [inline doc/nrf/libraries/modem/modem_key_mgmt.rst:109]: nit: some prints have `\n` and some don't. either is fine but align for consistency

**Review by lemrey — APPROVED** [review]
Thank you

File: doc/nrf/libraries/modem/modem_key_mgmt.rst (hunk 2 above)
Comment (eivindj-nordic) [reply doc/nrf/libraries/modem/modem_key_mgmt.rst:109]: Added `\n` to all.

File: doc/nrf/libraries/modem/modem_key_mgmt.rst [hunk 3]
```diff
@@ -57,6 +57,82 @@ The PEM format looks like this::
...
+
+   int err;
+   char cert[CERT_SIZE];
+
+   err = modem_key_mgmt_read(<YOUR_SEC_TAG>, MODEM_KEY_MGMT_CRED_TYPE_CA_CHAIN, cert, sizeof(cert));
+   if (err) {
+           printk("Failed to read certificate\n");
+   }
+
+The following code snippet shows how to deleta a CA chain certificate stored in the modem:
```
Comment (divipillai) [inline doc/nrf/libraries/modem/modem_key_mgmt.rst:126]:
```suggestion
The following code snippet shows how to delete a CA chain certificate stored in the modem:
```

Comment (eivindj-nordic) [reply doc/nrf/libraries/modem/modem_key_mgmt.rst]: Thanks, updated.

**Review by divipillai — APPROVED** [review]

**Comment by glarsennordic** [comment]
BTW, I just noticed this, but: It seems `modem_key_mgmt_read` expects a ***pointer*** to a size, rather than a size directly -- but the example showing `modem_key_mgmt_read` just passes `sizeof` directly?

Not sure whether this applies to the other examples. Or maybe I am misunderstanding something.

Leaving a comment instead of filing a ticket because I was investigating something else and just happened to be in the area and noticed this, and its entirely possible I just fundamentally misunderstand something

**Comment by eivindj-nordic** [comment]
> BTW, I just noticed this, but: It seems `modem_key_mgmt_read` expects a _**pointer**_ to a size, rather than a size directly -- but the example showing `modem_key_mgmt_read` just passes `sizeof` directly?
> 
> Not sure whether this applies to the other examples. Or maybe I am misunderstanding something.
> 
> Leaving a comment instead of filing a ticket because I was investigating something else and just happened to be in the area and noticed this, and its entirely possible I just fundamentally misunderstand something

Thanks @glarsennordic You are correct. I opened #17636.

**Pull request merged**